File: src/app/models/Project.ts
import { User } from './User';
import { Task } from './Task';

export class Project {
    projectId: number;
    projectName: string;
    projectStatus: number;
    userDTO: User;
    tasksDTO: Task [];



    constructor(projectId: number, projectName: string, projectStatus: number, userDTO: User, tasksDTO: Task [] ) {
        this.projectId = projectId;
        this.projectName = projectName;
        this.projectStatus = projectStatus;
        this.userDTO = userDTO;
        this.tasksDTO = tasksDTO;
    }




}

File: src/main/java/it/contrader/wmesspring/model/Project.java
package it.contrader.wmesspring.model;

import java.io.Serializable;
import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;


/**
 * The persistent class for the projects database table.
 * 
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="projects")
@NamedQuery(name="Project.findAll", query="SELECT p FROM Project p")
public class Project implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="project_id")
	private int projectId;

	@Column(name="project_name")
	private String projectName;

	@Column(name="project_status")
	private int projectStatus;

	//bi-directional many-to-one association to User
	@ManyToOne
	@JoinColumn(name="user_id")
	private User user;

	@OneToMany(mappedBy="project")
	private List<Task> projectsTasks;

}